### PR TITLE
Package zoltan

### PR DESCRIPTION
**Pull request by SylvainCorlay**
This is a dependency of Proteus.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
cc @minrk this package makes use of the mpi providers and I am not 100% sure that I am doing this right.

File: recipes/zoltan/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{% set name = "zoltan" %}
+{% set version = "3.83" %}
+{% set build = 0 %}
+
+{% set mpi_prefix = "mpi_" + mpi %}
```
[the remark this answers is not in the document]
Comment (SylvainCorlay) [reply recipes/zoltan/meta.yaml]: fixed! thanks @minrk

File: recipes/zoltan/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+about:
+  home: http://www.cs.sandia.gov/Zoltan/
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: COPYRIGHT_AND_LICENSE
+  summary: Parallel Partitioning, Load Balancing and Data-Management Services
+  description: |
+    The Zoltan Library provides critical data-management services to a wide range of parallel applications.
+  doc_url: https://github.com/simplejson/simplejson
+  dev_url: https://github.com/simplejson/simplejson
```
[the remark this answers is not in the document]
Comment (SylvainCorlay) [reply recipes/zoltan/meta.yaml]: thx

File: recipes/zoltan/build.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+mkdir build; cd $_
+
+../configure \
+    --prefix=$PREFIX \
+    --with-cflags=-fPIC \
+    --with-cxxflags=-fPIC
```
[the remark this answers is not in the document]
Comment (SylvainCorlay) [reply recipes/zoltan/build.sh]: resolved

File: recipes/zoltan/conda_build_config.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+mpi:
+  - mpich
+#  - openmpi
```
Comment (scopatz) [inline recipes/zoltan/conda_build_config.yaml]: Is this file needed?

**Comment by SylvainCorlay** [comment]
This was picked up by @davidbrochart in #9796. Closing.
